pkg/task/restore.go

```diff
@@ -261,68 +240,80 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 	}
 	batchSize = utils.MinInt(batchSize, maxRestoreBatchSizeLimit)
 
-	tiflashStores, err := conn.GetAllTiKVStores(ctx, client.GetPDClient(), conn.TiFlashOnly)
+	// Redirect to log if there is no log file to avoid unreadable output.
+	updateCh := g.StartProgress(
+		ctx,
+		cmdName,
+		// Split/Scatter + Download/Ingest + Checksum
+		int64(restore.EstimateRangeSize(files)+len(files)+len(tables)),
+		!cfg.LogProgress)
+	defer updateCh.Close()
+	sender, err := restore.NewTiKVSender(ctx, client, updateCh)
 	if err != nil {
-		return errors.Trace(err)
-	}
-	rejectStoreMap := make(map[uint64]bool)
-	for _, store := range tiflashStores {
-		rejectStoreMap[store.GetId()] = true
+		return err
 	}
+	manager := restore.NewBRContextManager(client)
```

Comment:
[L218-L225](https://github.com/pingcap/br/pull/266/files#diff-bab7dfeaf86ea3b3e8c02d466857ffb8R218-R225) can be simplify, by the way can we include `RestorePreWork` `RestorePostWork` into a new manager?
when  Enter when Batcher created and Leave when `Batcher.Close()`.

Reply:
Yes, but let me merge the master to this firstly...

Reply:
We maybe need to add a method like `AllDone` to `ContextManager` and call it when batcher closed.